### PR TITLE
Storybook: Adds FormToggle component

### PR DESCRIPTION
**Pull request by enejb**
## Description
Adds FormToggle component to storybook

## How has this been tested?
Run `npm run design-system:dev` and confirm component shows up under Form Toggle
Click on the toggle does it work as expected?

## Screenshots
Screen Shot 2019-10-23 at 9 15 55 PM

## Types of changes
Storybook

### CODE REVIEW

**Comment by ItsJonQ** [comment]
@enejb Awesome! Thank you for creating this and helping out with Storybook!

It looks good to me 👍

File: packages/components/src/form-toggle/stories/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Internal dependencies
+ */
+import FormToggle from '../';
+
+export default { title: 'Form Toggle', component: FormToggle };
+
+export const _default = () => {
+	const [ isChecked, setChecked ] = useState( true );
```
Comment (gziolo) [inline packages/components/src/form-toggle/stories/index.js]: The same comment as in https://github.com/WordPress/gutenberg/pull/18070#discussion_r339419618:

> According to documentation, the function exported as a story isn't a React component. It means that we shouldn't use hooks in this context as they might stop to work in the near future if they change the way all that works.
> 
> The easiest way to get rid of this warning is to move the part which manages state out of this function. See a similar usage:
> https://github.com/WordPress/gutenberg/blob/a6e1252886e8d990627ac4bf4796f768bfad2a44/packages/components/src/checkbox-control/stories/index.js#L18-L28

Related docs section: https://storybook.js.org/docs/formats/component-story-format/#story-exports

> By default every named export in the file represents a story function.
> 
> Story functions can be annotated with a story object to define story-level decorators and parameters, and also to define the name of the story.

If you would name this export differently, lint rules related to React hooks would warn as well, as they enforce to be inside a React component (the name needs to start with uppercase).

@ItsJonQ @mkaz - do you think we should be more strict about it? I know that the StoryShots integration I explored will error when you use React hooks in story functions.

**Review by gziolo — COMMENTED** [review]
@enejb great work jumping into creating a story 🥇 

I have a more general question in regards to using React hooks in the exports. I saw the same approach in the past and I see it in other PRs so I think we should clarify how we tackle it first.

File: packages/components/src/form-toggle/stories/index.js (hunk 1 above)
Comment (ItsJonQ) [reply packages/components/src/form-toggle/stories/index.js]: @gziolo After creating a couple of stories, I started doing something like this:

```jsx
...

const Example = () => {
	const [ isChecked, setChecked ] = useState( true );
	return (
		<FormToggle
			checked={ isChecked }
			onChange={
				() => {
					setChecked( ! isChecked );
				}
			}
		/>
	);
}

export const _default = () => {
	return <Example />
};
```

This bypasses the React Hook issue that StoryShots/Book complains about.

What do you think @enejb ? :D

**Comment by mkaz** [comment]
@ItsJonQ My only issue with the `<Example/>` export you showed is the View Code from the Docs tab isn't quite complete. If you view the source from the Story tab, its fine, so not the biggest deal.

Here's a screen shot of what it looks like in ScrollLock.

Screenshot from 2019-10-28 11-03-39

**Comment by ItsJonQ** [comment]
@mkaz Oh dang! That's a bit of a bummer... :(

**Comment by enejb** [comment]
Thanks for the reviews! 
Let me know if there is anything else that I missed.

**Review by gziolo — APPROVED** [review]
Nice, thanks for the refactor. Looks great now 💯

File: packages/components/src/form-toggle/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import FormToggle from '../';
+
+export default { title: 'Form Toggle', component: FormToggle };
```
Comment (gziolo) [inline packages/components/src/form-toggle/stories/index.js]:
```suggestion
export default { title: 'FormToggle', component: FormToggle };
```

Comment (gziolo) [reply packages/components/src/form-toggle/stories/index.js]: I don't have perms to commit myself. We can tackle it separately. It's just me looking for consistency in naming :)

**Review by gziolo — CHANGES_REQUESTED** [review]
One more thing which I discovered while testing:

Screen Shot 2019-10-29 at 07 57 45

We should add the label and it can be integrated with knobs, see:
https://github.com/WordPress/gutenberg/blob/f56110d0210daaad7499611cfc822f33595060b7/packages/components/src/external-link/stories/index.js#L13
or
https://github.com/WordPress/gutenberg/blob/f56110d0210daaad7499611cfc822f33595060b7/packages/components/src/checkbox-control/stories/index.js#L31

File: packages/components/src/form-toggle/stories/index.js (hunk 2 above)
Comment (ItsJonQ) [reply packages/components/src/form-toggle/stories/index.js]: Following up on this! We've tidied up Storybook a bit. For components, we would label it like:

`Components|FormToggle`

Which automatically categorizes it in Storybook's sidebar

**Review by gziolo — APPROVED** [review]
I added a commit which formats file with Prettier.

Let's get it in and iterate. @enejb, thank you for doing your part :)

**Pull request merged**